AnyText/AnyText.Core/Parser.FoldingRanges.cs

```diff
@@ -0,0 +1,180 @@
+                        StartCharacter = (uint)commentRuleApplication.CurrentPosition.Col,
+                        EndLine = (uint)(endCommentRuleApplication.CurrentPosition.Line),
+                        EndCharacter = 0, // determining the end character using length or examinedTo is inconsistent and can lead to wrap around when casting to uint
+                        Kind = "comment"
+                    };
+
+                    result.Add(commentsFoldingRange);
+                }
+            }
+
+            return result.ToArray();
+        }
+
+        private ParsedFoldingRange GetImportsFoldingRange(RuleApplication ruleApplication)
+        {
+            var firstInner = (ruleApplication as MultiRuleApplication).Inner.First();
+            var lastInner = (ruleApplication as MultiRuleApplication).Inner.Last();
+
+            var importsFoldingRange = new ParsedFoldingRange()
+            {
+                StartLine = (uint)firstInner.CurrentPosition.Line,
+                StartCharacter = (uint)firstInner.CurrentPosition.Col,
+                EndLine = (uint)lastInner.CurrentPosition.Line,
+                EndCharacter = 0, // determining the end character using length or examinedTo is inconsistent and can lead to wrap around when casting to uint
+                Kind = "imports"
+            };
+
+            return importsFoldingRange;
+        }
+
+        private ParsedFoldingRange GetRegionFoldingRange(MultiRuleApplication multiRuleApplication)
+        {
+            var firstInner = multiRuleApplication.Inner.First();
+            var lastInner = multiRuleApplication.Inner.Last();
+
+            var regionFoldingRange = new ParsedFoldingRange()
+            {
+                StartLine = (uint)firstInner.CurrentPosition.Line,
+                StartCharacter = (uint)firstInner.CurrentPosition.Col,
+                EndLine = (uint)lastInner.CurrentPosition.Line,
+                EndCharacter = 0, // determining the end character using length or examinedTo is inconsistent and can lead to wrap around when casting to uint
+                Kind = "region"
+            };
+
+            return regionFoldingRange;
+        }
+
+        private ParsedFoldingRange GetFoldingRange(MultiRuleApplication multiRuleApplication)
+        {
+            var firstInner = multiRuleApplication.Inner.First();
+            var lastInner = multiRuleApplication.Inner.Last();
+
+            var foldingRange = new ParsedFoldingRange()
+            {
+                StartLine = (uint)firstInner.CurrentPosition.Line,
+                StartCharacter = (uint)firstInner.CurrentPosition.Col,
+                EndLine = (uint)lastInner.CurrentPosition.Line,
+                EndCharacter = 0 // determining the end character using length or examinedTo is inconsistent and can lead to wrap around when casting to uint
+            };
+
+            return foldingRange;
+        }
+
+        public class ParsedFoldingRange
```

Comment:
Bitte in eine eigene Datei und mit Dokumentation und Properties statt Feldern (automatische Properties sind aber ok)